example.tf

```diff
@@ -126,18 +125,18 @@ resource "aws_instance" "os_win_agent" {
     create = "15m"
   }
 
-  connection {
-    host = coalesce(self.public_ip, self.private_ip)
-    type = "winrm"
-    user = "Administrator"
-    password = rsadecrypt(self.password_data, file("~/.ssh/id_rsa"))
-    timeout = "10m"
-  }
-
   provisioner "puppet" {
     open_source = true
     server = aws_instance.puppetmaster.public_dns
     server_user = "ubuntu"
+
+    connection {
+      host = coalesce(self.public_ip, self.private_ip)
+      type = "winrm"
+      user = "Administrator"
+      password = rsadecrypt(self.password_data, file("~/.ssh/id_rsa"))
+      timeout = "10m"
```

Comment:
Note that now I've been through the code I don't believe this timeout here does anything ... ?